File: careercup/interleaved.py
'''Three strings say A,B,C are given to you. Check weather 3rd string is interleaved from string A and B. 
Ex: A="abcd" B="xyz" C="axybczd". answer is yes. o(n)
https://www.careercup.com/question?id=14539805
'''

def interleaved(a, b, c):
	clist = list(c)
	for l in a:
		if l in clist:
			clist.remove(l)
	print "".join(clist)
	if b in "".join(clist):
		return True
	else:
		return False
A="abcd"
B="xyz"
C="axybczd"
print interleaved(A, B, C)

File: careercup/interleaved.c
#include <stdio.h>

char * interleaved(char a[], char b[], char c[]) {
	int ia = 0, ib = 0, ic;
	for (ic = 0; c[ic]; ic++) {
		if (c[ic] == a[ia])
			ia++;
		else if (c[ic] == b[ib])
			ib++;
		else
			return "no";
	}
	return "yes";
}

int main(int argc, const char *argv[]) {
	printf("%s\n", interleaved("abcd", "xyz", "axybczd"));
	return 0;
}

